pkg/v1/providers/tests/unit/avi_test.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package unit
+
+import (
+	"io"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/providers/tests/unit/matchers"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/providers/tests/unit/ytt"
+)
+
+type yttValues map[string]string
+
+func (v yttValues) toReader() io.Reader {
+	return strings.NewReader(createDataValues(v))
+}
+
+func (v yttValues) Set(key, value string) {
+	v[key] = value
+}
+
+func (v yttValues) Delete(key string) {
+	delete(v, key)
+}
+
+func (v yttValues) DeepCopy() yttValues {
+	other := make(yttValues)
+	for key, value := range v {
+		other[key] = value
+	}
+	return other
+}
+
+func assertNotFound(docs []string, err error) {
+	Expect(err).NotTo(HaveOccurred())
+	Expect(docs).To(HaveLen(0))
+}
+
+func assertFoundOne(docs []string, err error) {
+	Expect(err).NotTo(HaveOccurred())
+	Expect(docs).To(HaveLen(1))
+}
+
+const (
+	AviUsername            = "admin"
+	AviPassword            = "pass_1234"
+	AviCaName              = "avi-controller-ca"
+	AviCaDataB64           = "LS0tLS1CR...LS0tLQo="
+	AviAdminCredentialName = "avi-controller-credentials"
+)
+
+func NoAVIRelatedObjects(output string) {
+	assertNotFound(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "Secret",
+		"$.metadata.name": "avi-secret",
+	}))
+	assertNotFound(matchers.FindDocsMatchingYAMLPath(output, map[string]string{
+		"$.kind":          "ServiceAccount",
+		"$.metadata.name": "ako-sa",
```

Comment:
Question: would it be a good idea to define constants for this field and other similar fields ? So that it would be easier to maintain.

Reply:
that's definitely a point. I will change them to constant if I used them more than twice in the following PRs. But so far it's good enough IMO